src/mlpack/core/cv/simple_cv_impl.hpp

```diff
@@ -0,0 +1,208 @@
+         typename PredictionsType,
+         typename WeightsType>
+MLAlgorithm& SimpleCV<MLAlgorithm,
+                      Metric,
+                      MatType,
+                      PredictionsType,
+                      WeightsType>::Model()
+{
+  if (modelPtr == nullptr)
+    throw std::logic_error(
+        "SimpleCV::Model(): attempted to access an uninitialized model");
+
+  return *modelPtr;
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<typename DataArgsTupleT,
+         typename>
+void SimpleCV<MLAlgorithm,
+              Metric,
+              MatType,
+              PredictionsType,
+              WeightsType>::Init(const double validationSize,
+                                 const DataArgsTupleT& dataArgsTuple)
+{
+  xs = std::get<0>(dataArgsTuple);
+  ys = std::get<1>(dataArgsTuple);
+
+  Base::AssertDataConsistency(xs, ys);
+
+  InitTrainingAndValidationSets(validationSize);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<typename DataArgsTupleT,
+         typename,
+         typename>
+void SimpleCV<MLAlgorithm,
+              Metric,
+              MatType,
+              PredictionsType,
+              WeightsType>::Init(const double validationSize,
+                                 const DataArgsTupleT& dataArgsTuple)
+{
+  xs = std::get<0>(dataArgsTuple);
+  ys = std::get<1>(dataArgsTuple);
+  weights = std::get<2>(dataArgsTuple);
+
+  Base::AssertDataConsistency(xs, ys, weights);
+
+  InitTrainingAndValidationSets(validationSize);
+
+  trainingWeights = GetSubset(weights, 0, trainingXs.n_cols - 1);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+void SimpleCV<MLAlgorithm,
+              Metric,
+              MatType,
+              PredictionsType,
+              WeightsType>::InitTrainingAndValidationSets(
+    const double validationSize)
+{
+  size_t numberOfTrainingPoints = CalculateAndAssertNumberOfTrainingPoints(
+      validationSize);
+
+  trainingXs = GetSubset(xs, 0, numberOfTrainingPoints - 1);
+  trainingYs = GetSubset(ys, 0, numberOfTrainingPoints - 1);
+
+  validationXs = GetSubset(xs, numberOfTrainingPoints, xs.n_cols - 1);
+  validationYs = GetSubset(ys, numberOfTrainingPoints, xs.n_cols - 1);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+size_t SimpleCV<MLAlgorithm,
+                Metric,
+                MatType,
+                PredictionsType,
+                WeightsType>::CalculateAndAssertNumberOfTrainingPoints(
+    const double validationSize)
+{
+  if (validationSize < 0.0F || validationSize > 1.0F)
```

Comment:
I agree, it's not clear, it should be:

> To conserve lines, refrain from using braces if the if, while, or for statement applies to a line statement.

There are definitely other places in the codebase, where we do something similar, I try to comment on it once I see something like that. I will update the style section to make it more clear, and adjust the lines afterwards, so you can leave it as it is if you like.